configure.ac

```diff
@@ -43,18 +43,32 @@ fi
 # Check for database support.
 AC_ARG_WITH(database, AC_HELP_STRING([--with-database=[[mysql,mariadb]]],
 			[Include support for the given database management system]),
-			database=$withval, database=mariadb)
-if test x"$database" != xno -a x"$database" != xmariadb -a x"$database" != xmysql ; then
+			database=$withval, database=auto)
+if test x"$database" != xno -a x"$database" != xauto -a x"$database" != xmariadb -a x"$database" != xmysql ; then
 	AC_MSG_ERROR([$database is not a valid value for --with-database])
 	database = "no"
 fi
 
-AC_ARG_WITH(database-path, AC_HELP_STRING([--with-database-path=/folder/containing/[[[mysql,mariadb]]]_config], [Path to the utility tool providing build configuration for the selected DBMS. If not set, lookup in PATH]),
+AC_ARG_WITH(database-path, AC_HELP_STRING([--with-database-path=/folder/containing/mysql_config], [Path to the utility tool providing build configuration for the selected DBMS. If not set, lookup in PATH]),
```

Comment:
Something is odd with `database-path` since it is now relevant only for MySQL. It was initially meant to be database-neutral but that does not work anymore with your proposal.

Thinking about it again, I now believe it would be clearer to use two configure parameters:
- `--with-mysql` will retain the meaning it has always had. An optional value sets the path to the mysql_config utility script
- `--with-mariadb` will be added, without value and mariadb is configured using pkgconfig as you suggested

Those two parameters should be mutually exclusive to avoid link problems. In terms of configuration, it should be possible to have one service for mariadb and another one for mysql but since mysql and mariadb libraries provide homonymous symbols, I'm afraid we can't link with the two of them simultaneously.

Reply:
Exactly, database-path is only relevant for MySQL. Maybe it could be renamed to --with-mysql-path to make it more clear.
I still like your idea with --with-database and using the generic DBMS values as it forces to decide between them if both should be available. I could imagine this could happen more often in the future as many projects will have to migrate from MySQL to MariaDB. With your initial idea you prepare the linknx project very well for that change. I suggest to go for it.

Reply:
I would like to come back to two independent parameters (`--with-mysql` and `--with-mariadb`) because the current proposal suggests that there can be support for only one DBMS at a time. MariaDB and MySQL are mutually exclusive but other DBMS could coexist, so the way I defined it in my PR is not consistent.

I am merging this PR anyway, as your suggestion to use pkgconfig for MariaDB still makes sense.